src/main/java/com/synopsys/integration/detect/tool/signaturescanner/BlackDuckSignatureScannerTool.java

```diff
@@ -63,71 +63,90 @@ public BlackDuckSignatureScannerTool(BlackDuckSignatureScannerOptions signatureS
 
     // TODO: Don't accept an Optional as a parameter.
     public SignatureScannerToolResult runScanTool(CodeLocationCreationService codeLocationCreationService, BlackDuckServerConfig blackDuckServerConfig, NameVersion projectNameVersion, Optional<File> dockerTar) throws DetectUserFriendlyException {
-        DetectConfigurationFactory detectConfigurationFactory = detectContext.getBean(DetectConfigurationFactory.class);
         ConnectionFactory connectionFactory = detectContext.getBean(ConnectionFactory.class);
         DirectoryManager directoryManager = detectContext.getBean(DirectoryManager.class);
         CodeLocationNameGenerator codeLocationNameService = detectContext.getBean(CodeLocationNameGenerator.class);
         CodeLocationNameManager codeLocationNameManager = detectContext.getBean(CodeLocationNameManager.class, codeLocationNameService);
 
+        Optional<Path> localScannerInstallPath = determineLocalScannerInstallPath();
+
+        ExecutorService executorService = Executors.newFixedThreadPool(signatureScannerOptions.getParallelProcessors());
+        IntEnvironmentVariables intEnvironmentVariables = IntEnvironmentVariables.includeSystemEnv();
+        ScanBatchRunnerFactory scanBatchRunnerFactory = new ScanBatchRunnerFactory(intEnvironmentVariables, executorService);
```

Comment:
While these objects aren't from DetectContext, here again we're creating stuff instead of asking for it to be given to us.